sympy/core/add.py

```diff
@@ -339,21 +339,18 @@ def as_coeff_add(self, *deps):
         (0, (7*x,))
         """
         if deps:
-            l1 = []
```

Comment:
I saw someone, perhaps @oscarbenjamin , do this like
```python
L = l1, l2 = [], []
for f in self.args:
    L[f.has(*deps)].append(f)
return self._new_rawargs(*l2), tuple(l1)
```
Since this is a pretty low level function would it make sense to do that and avoid another function call? (Inside `sift` the same thing is being done so all that would be saved is the function call.)

Reply:
Could be. I think the main performance hog is `as_independent` though, and to some extent hiding things into sift is a bit cleaner compared to that code. But then I have no idea how much overhead there is for function calls.